ReSharper.FSharp/src/FSharp.TypeProviders.Host/src/Cache/SharedProvidedCache.cs

```diff
@@ -57,22 +68,91 @@ public bool TryGetKey(T model, int requestingTypeProviderId, out int key)
         return true;
       });
 
-    public abstract string Dump();
+    public virtual string Dump() => "";
+
+    public IEnumerator<KeyValuePair<T, (int id, HashSet<int> referencingProviders)>> GetEnumerator() =>
+      IdsCache.GetEnumerator();
+
+    IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
   }
 
-  public class ProvidedTypesCache : SharedProvidedCache<ProvidedType>
+  /// Cache holding types:
+  /// 1. Created by particular type provider instances.
+  /// 2. Coming from referenced assemblies metadata and not depending on particular type provider instances.
+  ///
+  /// Sharing created by provider types is not allowed, since type provider instantiations
+  /// depend on project configuration and may produce different types with the same name.
```

Comment:
```suggestion
  /// Sharing types created by type providers is not allowed, since type provider instantiations
  /// depend on project configuration and may produce different types with the same name.
```